### PR TITLE
Make auth tests work with async/await

### PR DESCRIPTION
**Pull request by joyeecheung**
Fixes: https://github.com/joyeecheung/node-core-utils/issues/32

### CODE REVIEW

**Comment by joyeecheung** [comment]
cc @cPhost @addaleax @refack

File: test/unit/auth.test.js [hunk 1]
```diff
@@ -33,80 +36,98 @@ describe('auth', () => {
...
-    { env: Object.assign({}, process.env, { HOME }) });
-  let stderr = '';
-  proc.stderr.setEncoding('utf8');
-  proc.stderr.on('data', (chunk) => { stderr += chunk; });
-  proc.on('close', () => {
-    if (stderr) { throw new Error(`unexpected stderr:\n${stderr}`); }
-  });
+    const proc = spawn(process.execPath,
+      [ require.resolve('../fixtures/run-auth') ],
+      // XXX this could just be env: { ...process.env, HOME } but the test loader
```
Comment (joyeecheung) [inline test/unit/auth.test.js]: Ah we already know why this happens in https://github.com/joyeecheung/node-core-utils/pull/12#issuecomment-339955478 ...Can remove this now.

Comment (addaleax) [reply test/unit/auth.test.js]: Yup… it’s a shame though :smile:

**Comment by cPhost** [comment]
Great! and that closes https://github.com/issues/32

File: test/unit/auth.test.js (hunk 1 above)
Comment (joyeecheung) [reply test/unit/auth.test.js]: @addaleax We'll just have to wait until 2018 (March?), I guess

Comment (addaleax) [reply test/unit/auth.test.js]: Is that the acorn release schedule or why March? ^^

Comment (addaleax) [reply test/unit/auth.test.js]: like, this works in Node 8 & 9 without a flag, I’m surprised userland parsers haven’t picked it up before V8…

Comment (joyeecheung) [reply test/unit/auth.test.js]: @addaleax IIRC ECMA has an assembly that approve the spec in March so by Feb they would need to cut releases before submitting them.

Comment (joyeecheung) [reply test/unit/auth.test.js]: @addaleax Acorn has a few plugins that support the object spread syntax already, they are just not included by default, I guess they are waiting for the spec to drop.

**Comment by cPhost** [comment]
That should be fixed now.

**Comment by refack** [comment]
Wow you quick 😄

**Pull request merged**